### PR TITLE
Add Pushover Receiver type to AlertmanagerConfig CRD

### LINKED ISSUE #3538: Add Pushover Receiver type to AlertmanagerConfig CRD
**What is missing?**

In #3451 we added the AlertmanagerConfig CRD, with just the PagerDuty receiver type, meaning that alert notifications can only be routed to PagerDuty. Alertmanager supports several other Receiver types, with this being one (other issues will be created for the others).

**Why do we need it?**

We should be able to define all of the Alertmanager receivers in AlertmanagerConfig CRs.

**Anything else we need to know?**:

The proposal for AlertmanagerConfig CRD is [here](https://docs.google.com/document/d/1aVVttvocop8zNezwrNFbHh_HKRW_fhJUZqhmrr8fGK0/edit#heading=h.5ua2yyj2xm4h), and it includes some information on this receiver type.

The Alertmanager config docs [here](https://prometheus.io/docs/alerting/latest/configuration/) show how this receiver is expected to be in the resulting Alertmanager config.

### PR DESCRIPTION
**Pull request by sev3ryn**
fixes #3538

### CODE REVIEW

**Comment by simonpasquier** [comment]
Thanks! cc @grdryn

**Review by grdryn — APPROVED** [review]
LGTM, thanks @sev3ryn! :+1:

**Comment by lilic** [comment]
Same as other one sadly @sev3ryn there are conflicts.

File: pkg/alertmanager/operator.go [hunk 1]
```diff
@@ -1083,6 +1088,48 @@ func checkWechatConfigs(ctx context.Context, configs []monitoringv1alpha1.WeChat
...
+			}
+		}
+		if config.Token != nil {
+			if _, err := store.GetSecretKey(ctx, namespace, *config.UserKey); err != nil {
+				return err
+			}
+		}
+
+		if config.URL != nil {
+			_, err := url.Parse(*config.URL)
```
Comment (simonpasquier) [inline pkg/alertmanager/operator.go:1106]: `URL` can be a template string so we shouldn't check its validity.

File: pkg/alertmanager/operator.go [hunk 2]
```diff
@@ -1083,6 +1088,48 @@ func checkWechatConfigs(ctx context.Context, configs []monitoringv1alpha1.WeChat
 	return nil
 }
 
+func checkPushoverConfigs(ctx context.Context, configs []monitoringv1alpha1.PushoverConfig, namespace string, key string, store *assets.Store) error {
+	for i, config := range configs {
+
+		if config.UserKey != nil {
```
Comment (simonpasquier) [inline pkg/alertmanager/operator.go:1094]: `UserKey` is mandatory

File: pkg/alertmanager/operator.go [hunk 3]
```diff
@@ -1083,6 +1088,48 @@ func checkWechatConfigs(ctx context.Context, configs []monitoringv1alpha1.WeChat
...
+			if _, err := store.GetSecretKey(ctx, namespace, *config.UserKey); err != nil {
+				return err
+			}
+		}
+		if config.Token != nil {
```
Comment (simonpasquier) [inline pkg/alertmanager/operator.go:1099]: Same for `Token`

**Comment by sev3ryn** [comment]
@simonpasquier fixed all your suggestions - see 53d94a68

**Review by simonpasquier — APPROVED** [review]

**Comment by simonpasquier** [comment]
:tada: many many thanks @sev3ryn for all the hard work! You deserve a big round of :clap: :clap: from the Prometheus operator community.

**Pull request merged**